### DISCUSSION TITLE
Inject Methods/Object with Plugins

### DISCUSSION THREAD

**Discussion by shadowtime2000** [eta-dev/eta#225] [Plugin ideas]
**Is your feature request related to a problem? Please describe.**
Currently there isn't a nice clean way to inject methods or objects with plugins for usage in the template.

**Describe the solution you'd like**
Some way to inject methods for usage.

**Additional context**
Carrying over discussion from Gitter, we could allow modification of prototype (I don't think that works), allowing injecting into `this` of template, or use some scoping tricks with dynamic closures to handle this (still thinking about it).

**Comment by shadowtime2000** [comment]
Possible usage:
```js
module.exports = {
	injectThis: (config) => {
		return {
			add: (a, b) => a + b
		}
	}
}
```

```ejs
<%= this.add(2, 3) %>
```

**Comment by lenovouser** [comment]
I'd really need this as well for e.g. global translation methods 👍🏻

**Comment by lenovouser** [comment]
@shadowtime2000 any updates or way one can help out with this?

**Reply by bgub** [reply]
@lenovouser we're close to releasing v3, but v3 will have almost the same plugin API as v2. Feel free to start working on a plugin that would do this!
